include/aspect/geometry_model/box.h

```diff
@@ -247,6 +261,11 @@ namespace aspect
          * The number of cells in each coordinate direction.
          */
         std::array<unsigned int, dim> repetitions;
+
+        /**
+         * Function to interpolate surface topography.
+         */
+        Functions::InterpolatedTensorProductGridData<dim-1> *surface_function;
```

Comment:
please only use smart pointers like `std::unique_ptr`, not raw pointers like this one. Smart pointer avoid problems of variable ownership and guarantee the memory is correctly freed when the variable is destroyed. See e.g. [here](https://github.com/geodynamics/aspect/blob/e9cd1c8fc83f0d5c6e52c9d775a71382a1597049/include/aspect/material_model/visco_plastic.h#L233) and [here](https://github.com/geodynamics/aspect/blob/e9cd1c8fc83f0d5c6e52c9d775a71382a1597049/source/material_model/visco_plastic.cc#L376) for how to use them.

Reply:
Done.